### PR TITLE
[Bugfix] Alternative view refactoring

### PR DESCRIPTION
**Pull request by ke4**
Manual checks

- [x] This branch is passing the CI tests as part of a pull request in the atlas-web-bulk or atlas-web-scxa repos and I have indicated that PR here.

### CODE REVIEW

File: src/main/java/uk/ac/ebi/atlas/trader/factory/BaselineExperimentFactory.java [hunk 1]
```diff
@@ -76,22 +83,39 @@ public BaselineExperiment create(ExperimentDto experimentDto,
...
-                                        "Related " +
-                                                experimentRepository.getPublicExperiment(altViewAccession).getType().getHumanDescription() + " experiment"
-
-                  )
-                    .collect(toImmutableList()));
+    private Collection<String>
+    extractAlternativeViewDescriptions(String experimentAccession, List<String> alternativeViewAccessions) {
+        final List<String> alternativeViewDescriptions = new ArrayList<>();
+
+        alternativeViewAccessions.forEach(alternativeAccession -> {
```
Comment (lingyun1010) [inline src/main/java/uk/ac/ebi/atlas/trader/factory/BaselineExperimentFactory.java:90]: Hi @ke4, I tested locally by manully editing the file `E-MTAB-2836-factor.xml` by adding `alternativeView` tag, as we don't have the correct version by downloading files from [ftp](http://ftp.ebi.ac.uk/pub/databases/arrayexpress/data/atlas/experiments/E-MTAB-2836/E-MTAB-2836-factors.xml). 

According to the data production team, the pair experiments which have `alternativeView` tags are `E-PROT-29` and `E-MTAB-2836`. However, on our local environment, we don't have `E-PROT-29`, so I just tested with some random experiments in our local database.

In one case, I added `E-PROT-39` as the `alternativeView`, and it works fine, showing `Related` as expected.

In another case, I added `E-MTAB-2909` as the `alternativeView` and it shows as below.

Screenshot 2024-04-09 at 16 20 01

The `See also` link is still working, but I suppose it should be `View by...` as they are the same experiment type. However, as experiment `E-MTAB-2909` doesn't contain any `factor.xml` file or the file doesn't contain any `defaultQueryFactorType` tag, it shows `Missing...` instead. Please check if it is supposed to be like this, or it's a bug logic thanks.

Comment (ke4) [reply src/main/java/uk/ac/ebi/atlas/trader/factory/BaselineExperimentFactory.java:90]: @lingyun1010 Thanks for this thorough testing. 👍

**Comment by ke4** [comment]
@lingyun1010 This is the new version as we discussed in our meeting this morning:
Screenshot 2024-04-10 at 12 47 36

File: src/main/java/uk/ac/ebi/atlas/trader/factory/BaselineExperimentFactory.java [hunk 2]
```diff
@@ -89,15 +89,18 @@ public BaselineExperiment create(ExperimentDto experimentDto,
...
                         :
                         "Related " + getExperimentType(alternativeAccession) + " experiment");
-            } catch (ResourceNotFoundException | XmlFileConfigurationException e) {
-
+            } catch (ResourceNotFoundException e) {
                 LOGGER.error("Experiment: {} has a "
                         + DESCRIPTION_FOR_MISSING_ALT_VIEW.toLowerCase() + alternativeAccession, experimentAccession);
                 alternativeViewDescriptions.add(DESCRIPTION_FOR_MISSING_ALT_VIEW + alternativeAccession);
+            } catch (XmlFileConfigurationException e) {
+                LOGGER.error("Experiment: {} has an alternative view {} that has a missing configuration file.",
```
Comment (lingyun1010) [inline src/main/java/uk/ac/ebi/atlas/trader/factory/BaselineExperimentFactory.java:101]: In our test case, `E-MTAB-2909` doesn't have a configuration file, but this XML exception doesn't catch it, and the description is `missing configuration file`, which is a bit confused.

Comment (ke4) [reply src/main/java/uk/ac/ebi/atlas/trader/factory/BaselineExperimentFactory.java:101]: @lingyun1010 What do you mean? Are you talking about the `log` statement? I think that should be OK. That is not appearing for the user. That is for us, for developers. ;-)

Comment (lingyun1010) [reply src/main/java/uk/ac/ebi/atlas/trader/factory/BaselineExperimentFactory.java:101]: Okay, make sense, then that's fair enough.

**Review by lingyun1010 — APPROVED** [review]
LGTM!

**Pull request merged**